frontend/src/lib/services/user-service.ts

```diff
@@ -95,4 +95,11 @@ export default class UserService extends APIService {
 		const res = await this.api.put(`/users/${id}/user-groups`, { userGroupIds });
 		return res.data as User;
 	}
+
+	async setUserAttribute(userId: string, attribute: string, value: any) {
+		const payload: Record<string, any> = {};
+		payload[attribute] = value;
+		const res = await this.api.put(`/users/${userId}/set`, payload);
+		return res.data as User;
+	}
```

Comment:
```suggestion

async setUserAttribute(userId: string, attribute: string, value: any): Promise<User> {
  const response = await this.api.put(`/users/${userId}/set`, { [attribute]: value });
  return response.data as User;
}
```

I think this solution is a bit cleaner.